src/software/ai/hl/stp/tactic/attacker/simulated_attacker_tactic_keep_away_test.cpp

```diff
@@ -0,0 +1,241 @@
+            while (ignore_score_checks)
+            {
+                yield("");
+            }
+
+            auto initial_enemy_risk_score = ratePassEnemyRisk(
+                enemy_team, pass, enemy_reaction_time, enemy_proximity_importance);
+
+            while (world_ptr->getMostRecentTimestamp() < Timestamp::fromSeconds(1))
+            {
+                yield("");
+            }
+
+            auto last_timestamp = world_ptr->getMostRecentTimestamp();
+
+            // test runs for NUM_CHECKS * CHECK_SCORE_INTERVAL seconds
+            for (int i = 0; i < NUM_CHECKS; i++)
+            {
+                // only check the enemy risk score every 0.5s to mitigate the "noise"
+                // inherent in the ratePass______ functions
+                while (world_ptr->getMostRecentTimestamp() - last_timestamp <
+                       CHECK_SCORE_INTERVAL)
+                {
+                    yield("");
+                }
+                last_timestamp = world_ptr->getMostRecentTimestamp();
+
+                // update the pass to reflect the new passer point, now that
+                // the ball has (probably) moved
+                Pass new_pass(world_ptr->ball().position(), pass.receiverPoint(),
+                              pass.speed());
+
+                auto current_enemy_risk_score =
+                    ratePassEnemyRisk(enemy_team, new_pass, enemy_reaction_time,
+                                      enemy_proximity_importance);
+
+                // make sure we improved over the initial enemy risk score
+                if (current_enemy_risk_score <= initial_enemy_risk_score)
+                {
+                    std::stringstream ss;
+                    ss << "At " << last_timestamp
+                       << " ratePassEnemyRisk didn't improve over initial! Went from "
+                       << initial_enemy_risk_score << " to " << current_enemy_risk_score;
+                    yield(ss.str());
+                }
+            }
+        },
+        [&](std::shared_ptr<World> world_ptr, ValidationCoroutine::push_type& yield) {
+            robotNotExcessivelyDribbling(1, world_ptr, yield);
+        },
+        [&](std::shared_ptr<World> world_ptr, ValidationCoroutine::push_type& yield) {
+            // test that the ball is always in the field boundaries
+            if (!contains(world_ptr->field().fieldLines(), world_ptr->ball().position()))
+            {
+                yield("Ball left the field boundaries!");
+            }
+            else
+            {
+                yield("");
+            }
+        }};
+
+    runTest(field, ball_state, friendly_robots, enemy_robots, {},
+            non_terminating_validation_functions, Duration::fromSeconds(3.0));
+}
+
+auto FIELD_TOP_LEFT = Field::createSSLDivisionBField().fieldLines().negXPosYCorner();
+
+INSTANTIATE_TEST_CASE_P(
+    PassEnvironment, SimulatedAttackerTacticKeepAwayTest,
+    ::testing::Values(
+        std::make_tuple(
```

Comment:
in this test, it looks like keepaway is direct the robot towards enemy robot 3 even though there's empty space in the -y direction

Reply:
did some parameter changes for marginally better perofrmance on this test case